### PR TITLE
Updating CircleCI config to be faster

### PR DESCRIPTION
**Pull request by aendrew**
This does a few things:

1. Removes some of the weird caching behaviour Callum added to counter some of the bugs in early versions of CircleCI v2
1. Uses `npm ci` to install exact deps using `package-lock.json`
1. Removes the `apt-get install` step, instead using the Dockerfile [here](https://github.com/aendrew/starter-kit-docker/blob/master/Dockerfile)

As a result CircleCI builds should be much, much faster. 

N.b., the Docker image I'm using is aendrew/starter-kit-docker:latest — I've namespaced it to my Docker/GitHub users because there's neither a Financial Times Dockerhub user nor one for ft-interactive. I figure I'll move the image and everything there eventually if and when those get created.

### CODE REVIEW

**Comment by aendrew** [comment]
Perf improvement:
- *Baseline* 2:07 224c4cdd399
- *This PR* 1:48 build time

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,54 +3,37 @@ jobs:
   build:
     working_directory: ~/project
     docker:
-      - image: node:8
+      - image: aendrew/starter-kit-docker:latest # https://github.com/aendrew/starter-kit-docker
```
Comment (joannaskao) [inline .circleci/config.yml]: Is this something we can create a fork of on the ft-interactive or Financial-Times organisations?

Comment (joannaskao) [reply .circleci/config.yml]: Oh wait, just read your note. Nevermind! Let's just make sure we remember to do this eventually.

**Comment by aendrew** [comment]
Let's not do this for the immediate future.

**Pull request closed without merging**